core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -611,7 +612,7 @@ private object RTS {
               // Interruption cannot be interrupted:
               this.noInterrupt += 1
 
-              curIo = IO.terminate[E, Any](die.get)
+              curIo = IO.terminate0[E, Any](Nil)
```

Comment:
This code is definitely wrong. I think it must be `IO.terminate0(status.get.causes)`.

Reply:
I made that change because in `kill0` the interruptions are added to `status.errors`, so when this action is evaluated by RTS, the errors will already be there and picked up in the `Terminate` branch. Or do you prefer that I don't add the errors in `kill0` and instead revert `shouldDie` to return an optional list of throwables? As far as I can see, the code above will only be executed after `kill0` has been called.

Reply:
Here's the [diff](https://github.com/scalaz/scalaz-zio/pull/107/commits/e954f3b0f3fe31dfe3498f1e5317bf77254307fd) for that particular change. Before that, I had the same exceptions reported twice in my test (only on Travis).